crates/sui-move-build/src/lib.rs

```diff
@@ -301,12 +322,19 @@ pub fn build_from_resolution_graph(
         }
         // TODO(https://github.com/MystenLabs/sui/issues/69): Run Move linker
     }
-    Ok(CompiledPackage {
+
+    let compiled_package = CompiledPackage {
         package,
         published_at,
         dependency_ids,
         bytecode_deps,
-    })
+    };
+
+    if with_unpublished_dependencies {
+        compiled_package.verify_unpublished_dependencies()?;
+    }
```

Comment:
This was also moved from `compile_package` in `sui/src/client_commands.rs` to here.